File: Python/94. Binary Tree Inorder Traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        result = []
        def dfs(node):
            if node.left: dfs(node.left)
            result.append(node.val)
            if node.right: dfs(node.right)
        if root: dfs(root)
        return result

File: C/94. Binary Tree Inorder Traversal.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
void inOrder(int* order, struct TreeNode* node, int* i) {
    if (node == NULL) return;

    inOrder(order, node->left, i);
    order[(*i)++] = node->val;
    inOrder(order, node->right, i);
}

int* inorderTraversal(struct TreeNode* root, int* returnSize){
    int* order = (int*)malloc(100 * sizeof(int));
    *returnSize = 0;
    inOrder(order, root, returnSize);
    return order;
}
